compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -109,26 +109,34 @@ const MAX_HASHED_BUFFER_LEN: usize = 2 * MAX_BYTES_TO_HASH;
 // large.
 impl hash::Hash for Allocation {
     fn hash<H: hash::Hasher>(&self, state: &mut H) {
+        let Self {
+            bytes,
+            provenance,
+            init_mask,
+            align,
+            mutability,
+            extra: _, // don't bother hashing ()
```

Comment:
Extra is not always `()`, so I think it is wrong to skip hashing the field altogether.

Reply:
Inside the impl block it is

Reply:
I'm also not sure it's worth adding any complexity to not hash `()`, does that completely optimize out?

Reply:
can you match it like `extra: ()`?

Reply:
great idea, I will.